services/src/main/java/services/TireServiceImpl.java

```diff
@@ -0,0 +1,101 @@
+package services;
+
+import cz.muni.fi.pa165.pneuservis.backend.dao.TireManagerDao;
+import cz.muni.fi.pa165.pneuservis.backend.entity.Tire;
+import cz.muni.fi.pa165.pneuservis.backend.entity.TireManufacturer;
+import cz.muni.fi.pa165.pneuservis.backend.entity.TireProperties;
+import cz.muni.fi.pa165.pneuservis.backend.enums.SeasonEnum;
+import cz.muni.fi.pa165.pneuservis.backend.enums.SpeedClassEnum;
+import cz.muni.fi.pa165.pneuservis.backend.enums.VehicleTypeEnum;
+import dto.TireManufacturerDTO;
+import dto.TirePropertiesDTO;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import javax.inject.Inject;
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+
+/***
+ * @author Jakub Palenik, 422453@mail.muni.cz
+ */
+
+@Service
+@Transactional
+public class TireServiceImpl implements TireService{
+
+    @Inject
+    private TireDataCache tdc;
+
+    @Inject
+    private TireManagerDao tireManagerDao;
+
+    @Override
+    public List<Tire> getAllTires() {
+        return tireManagerDao.retrieveAllTires();
```

Comment:
Shouldn't this be encapsulated in an unmodifiableList?

Reply:
true dat